### PR TITLE
TUP-20482 : active if with new legth strategy for comparing values.

### PR DESCRIPTION
**Pull request by akhabali**
This work still having issue with Tables

**What is the current behavior?** (You can also link to an open issue here)
https://jira.talendforge.org/browse/TUP-20482

**What is the new behavior?**

**Please check if the PR fulfills these requirements**

- [ ] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/model/parameter/TaCoKitElementParameter.java [hunk 1]
```diff
@@ -186,22 +186,22 @@ public void cloneAddionalInfoTo(final IAdditionalInfo targetAdditionalInfo) {
     public void setValue(final Object newValue) {
         final Object oldValue = super.getValue();
         super.setValue(newValue);
-        firePropertyChange(getName(), oldValue, newValue);
+        firePropertyChange("value", oldValue, newValue);
```
Comment (gonchar-ivan) [inline main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/model/parameter/TaCoKitElementParameter.java]: changing to "value" will break Suggestions method. see ActionParametersUpdater
However, we can consider to change logic in Suggestions or
we can file 2 events:
1. with getName()
2. with "value"

Comment (gonchar-ivan) [reply main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/model/parameter/TaCoKitElementParameter.java]: Aha, found a fix below. Thanks!

File: main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/ui/composite/TaCoKitComposite.java [hunk 2]
```diff
@@ -57,12 +54,11 @@
 
     private List<? extends IElementParameter> parameters;
 
-    private PropertyChangeListener paramChangedListener = new PropertyChangeListener() {
-
-        @Override
-        public void propertyChange(final PropertyChangeEvent evt) {
-            refresh();
+    private PropertyChangeListener redrawListener = evt -> {
+        if (!"show".equals(evt.getPropertyName())) {
```
Comment (gonchar-ivan) [inline main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/ui/composite/TaCoKitComposite.java]: I would suggest to rename "show" -> "refresh".
However, it is not critical

Comment (akhabali) [reply main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/ui/composite/TaCoKitComposite.java]: the property name is the name of the field that changed in the class that own the listeners and which is the field "show" here.

**Pull request merged**